sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py

```diff
@@ -44,15 +44,24 @@ class OpenTelemetrySpan(HttpSpanMixin, object):
     :paramtype kind: ~azure.core.tracing.SpanKind
     :keyword links: The list of links to be added to the span.
     :paramtype links: list[~azure.core.tracing.Link]
+    :keyword library_name: The name or identifier of the library that created the span.
+    :paramtype library_name: Optional[str]
+    :keyword library_version: The version of the library that created the span.
+    :paramtype library_version: Optional[str]
     """
 
     def __init__(self, span: Optional[Span] = None, name: str = "span", **kwargs: Any) -> None:
-        current_tracer = self.get_current_tracer()
 
         # TODO: Once we have additional supported versions, we should add a way to specify the version.
         self._schema_version = OpenTelemetrySchema.get_latest_version()
         self._attribute_mappings = OpenTelemetrySchema.get_attribute_mappings(self._schema_version)
 
+        current_tracer = trace.get_tracer(
+            kwargs.pop("library_name", None) or __name__,
```

Comment:
if we do `or __name__` we don't allow users to null out `library_name`, is that valid? Just want to double check on that

Reply:
A name is required by the get_tracer method in OpenTelemetry, so just trying to ensure something is passed in. 

Also, I need to rebase this PR.